pkg/controller/operators/olm/operator.go

```diff
@@ -158,6 +182,54 @@ func (a *Operator) syncDeployment(obj interface{}) (syncError error) {
 	return nil
 }
 
+func (a *Operator) syncOperatorGroups(obj interface{}) error {
+	op, ok := obj.(*v1alpha2.OperatorGroup)
+	if !ok {
+		log.Debugf("wrong type: %#v\n", obj)
+		return fmt.Errorf("casting OperatorGroup failed")
+	}
+
+	// write namespace matches to status field
+	selector, err := metav1.LabelSelectorAsSelector(&op.Spec.Selector)
+	if err != nil {
+		return err
+	}
+	operatorGroupOpts := metav1.ListOptions{LabelSelector: selector.String()}
+	namespaceList, err := a.OpClient.KubernetesInterface().CoreV1().Namespaces().List(operatorGroupOpts)
+	if err != nil {
+		return err
+	}
+	op.Status.Namespaces = namespaceList.Items
+
+	// make string list made up of comma separated namespaces
+	var b strings.Builder
+	nsCount := len(namespaceList.Items)
+	for i := 0; i < nsCount-1; i++ {
+		fmt.Fprintf(&b, "%v,", namespaceList.Items[i])
+	}
+	fmt.Fprintf(&b, "%v", namespaceList.Items[nsCount-1])
+
+	// write namespaces to watch in every deployment
+	csvsInNamespace := a.csvsInNamespace(op.GetNamespace())
+	for csvName, csv := range csvsInNamespace {
+		strategy, err := a.resolver.UnmarshalStrategy(csv.Spec.InstallStrategy)
+		if err != nil {
+			return fmt.Errorf("error unmarshaling strategy from ClusterServiceVersion '%s' with error: %s", csvName, err)
+		}
+
+		strategyDetailsDeployment, ok := strategy.(*install.StrategyDetailsDeployment)
+		if !ok {
+			return fmt.Errorf("could not assert strategy implementation as deployment for CSV %s", csvName)
+		}
+
+		for _, deploy := range strategyDetailsDeployment.DeploymentSpecs {
+			deploy.Spec.Template.Annotations["olm.targetNamespaces"] = b.String()
```

Comment:
Awesome, is this to have the CSV control loop update those CSVs'  deployments? If that's the case then we'll need to update those CSV's in Kube after we edit them. Also, I'm thinking that the deployment installer will need to check for updates to the annotations field; [possibly here](https://github.com/operator-framework/operator-lifecycle-manager/blob/05a838747633bb298e27ecf6209e25774ae0e034/pkg/controller/operators/olm/operator.go#L361:20) or [here](https://github.com/operator-framework/operator-lifecycle-manager/blob/05a838747633bb298e27ecf6209e25774ae0e034/pkg/controller/install/deployment.go#L201:39).